src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,45 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            mainDiagonal[i] = matrix[i][i];
+        }
+        return mainDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int depthIndex = matrix.length - 1;
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] = matrix[i][depthIndex];
+            depthIndex--;
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int minimal = matrix[0][0];
+        for (int[] innerMatrix: matrix) {
```

Comment:
```suggestion
        for (int[] innerMatrix : matrix) {
```
let's use Ctrl+Alt+L - it will add some white spaces